Students/Charlotte/ObjcTextInput/ObjcTextInput/Names List/TIIUIViewController.m

```diff
@@ -0,0 +1,135 @@
+#import "TIIViewModel.h"
+
+@interface TIIUIViewController () <UICollectionViewDataSource,UICollectionViewDelegate,TextDelegate>
+@property (nonatomic, weak) IBOutlet UICollectionView * collectionView;
+@property (nonatomic, strong) TIIViewModel * viewModel;
+@property (assign, nonatomic) BOOL isEditing;
+@property (nonatomic, strong) UILongPressGestureRecognizer * longPressGesture;
+@end
+
+@implementation TIIUIViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+    
+    self.viewModel = [[TIIViewModel alloc] initWithDefaultConfiguration];
+    
+    self.isEditing = NO;
+    
+    self.longPressGesture = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handleLongGesture:)];
+    [self.collectionView addGestureRecognizer:self.longPressGesture];
+
+    [self.collectionView reloadData];
+}
+
+#pragma mark - UILongPressGesture
+
+- (void)handleLongGesture:(UILongPressGestureRecognizer *)gesture {
+    NSIndexPath * selectedIndexPath;
+    switch (gesture.state) {
+        case UIGestureRecognizerStateBegan:
+            selectedIndexPath = [self.collectionView indexPathForItemAtPoint:[gesture locationInView:self.collectionView]];
+            
+            if (selectedIndexPath) {
+                [self.collectionView beginInteractiveMovementForItemAtIndexPath:selectedIndexPath];
+            }
+            break;
+        case UIGestureRecognizerStateChanged:
+            [self.collectionView updateInteractiveMovementTargetPosition:[gesture locationInView:self.collectionView]];
+            break;
+        case UIGestureRecognizerStateEnded:
+            [self.collectionView endInteractiveMovement];
+            break;
+        default:
+            [self.collectionView cancelInteractiveMovement];
+            break;
+    }
+}
+
+#pragma mark - UICollectionViewDataSource
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
+    return self.viewModel.numberOfItems;
+}
+
+- (__kindof UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+    static NSString * cellIdentifier = @"LabelCell";
+    
+    TIILabelCollectionViewCell * cell = [collectionView dequeueReusableCellWithReuseIdentifier:cellIdentifier forIndexPath:indexPath];
+    NSString * cellText = [self.viewModel objectForIndexPath:indexPath];
+    [cell configureWithString:cellText isInEditMode:self.isEditing];
+    return cell;
+}
+
+- (BOOL)collectionView:(UICollectionView *)collectionView canMoveItemAtIndexPath:(NSIndexPath *)indexPath {
+    return self.isEditing;
+}
+
+- (void)collectionView:(UICollectionView *)collectionView moveItemAtIndexPath:(NSIndexPath *)sourceIndexPath toIndexPath:(NSIndexPath *)destinationIndexPath {
+    [self.viewModel swapObjectAtIndexPath:sourceIndexPath withObjectAtIndexPath:destinationIndexPath];
+}
+
+#pragma mark - TextDelegate
```

Comment:
Mooi dat je 2 delegate methodes gemaakt hebt. Je zou ook een bool kunnen meegeven, new of niet. Maar dit vind ik eigenlijk een betere oplossing omdat je zo minder if else code moet schrijven.